doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,401 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.6
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(serve-deployment-graph)=
+
+# Deployment Graph
+
+```{note} 
+Note: This feature is still experimental in Alpha release, some APIs are subject to change.
+```
+
+## General Motivation
+
+Production machine learning serving pipelines are getting longer and wider. They often consist of multiple, or even tens of models collectively making a final prediction, such as image / video content classification and tagging, fraud detection pipeline with multiple policies and models, multi-stage ranking and recommendation, etc.
+
+Meanwhile, the size of a model is also growing beyond the memory limit of a single machine due to the exponentially growing number of parameters, such as GPT-3, sparse feature embeddings in recsys models such that the ability to do disaggregated and distributed inference is desirable and future proof.
+
+We want to leverage the programmable and general purpose distributed computing ability of Ray, double down on its unique strengths (scheduling, communication and shared memory) to facilitate authoring, orchestrating, scaling and deployment of complex serving pipelines under one set of DAG API, so a user can program & test multiple models or multiple shards of a single large model dynamically, deploy to production at scale, and upgrade individually.
+
+## Key requirements
+- Provide the ability to author a DAG of Serve nodes to form a complex inference graph.
+- Pipeline authoring experience should be fully Python-programmable with support for dynamic selection, control flows, user business logic, etc.
+- DAG can be instantiated and locally executed using tasks and actors API
+- DAG can be deployed as a group where individual nodes can be reconfigured and scaled indepenently.
+
+__[Full Ray Enhancement Proposal, REP-001: Serve Pipeline](https://github.com/ray-project/enhancements/blob/main/reps/2022-03-08-serve_pipeline.md)__
+
++++
+
+## Concepts
+
+### Deployment
+Upgradeable group of actors managed by the Serve controller. Currently the primary API in Ray Serve. At authoring time it’s the class or function under `@serve.deployment` decorator. 
+
+### Node
+Smallest unit in a graph, typically an annotated class but can also be a function, backed by a group of actors that are scalable and reconfigurable. 
```

Comment:
this is confusing, it doesn't explain that it's 1-1 with a deployment. How about combine the two and call it "Deployment Node" ?